### PR TITLE
Improve HTTP client

### PR DESCRIPTION
**Pull request by acabarbaye**
### Description

- Improved the HTTP configuration and selection to make it easier to use without having to jump into the code.
- Took the opportunity to improve the documentation of this module.
- Improved the retry policy and the handling of `Retry-After` header
- Improved the test coverage

### Test Coverage

- [x]  This change is covered by existing or additional automated tests.
- [ ]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [ ]  Additional tests are not required for this change (e.g. documentation update).

### CODE REVIEW

File: utils/filesystem/testdata/convmv [hunk 1]
```diff
@@ -0,0 +1,1195 @@
+#!/usr/bin/perl
```
Comment (SamCHogg) [inline utils/filesystem/testdata/convmv:1]: What's this? 👀

File: utils/http/configuration.go [hunk 2]
```diff
@@ -13,13 +13,42 @@ import (
 	"github.com/ARM-software/golang-utils/utils/config"
 )
 
+//HTTPClientConfiguration defines the client configuration. It can be used to tweak low level transport
+//parameters in order to adapt the client to your usecase.
+// If ensure about the values to set, use the DefaultHTTPClientConfiguration or FastHTTPClientConfiguration depending on
```
Comment (SamCHogg) [inline utils/http/configuration.go:18]:
```suggestion
// If unsure about the values to set, use the DefaultHTTPClientConfiguration or FastHTTPClientConfiguration depending on
```

File: utils/filesystem/testdata/convmv (hunk 1 above)
Comment (acabarbaye) [reply utils/filesystem/testdata/convmv:1]: a mistake
a tool for charset conversion

File: utils/http/interfaces.go [hunk 3]
```diff
@@ -2,17 +2,32 @@
  * Copyright (C) 2020-2022 Arm Limited or its affiliates and Contributors. All rights reserved.
  * SPDX-License-Identifier: Apache-2.0
  */
+
+// Package http provides a familiar HTTP client interface with
+// various implementations in order to answer common use cases: e.g. need for a robust client (performs retries on server errors with different retry policies depending on configuration),
+// for a fast client which leverages connection pools.
+// For the robust client, it is possible to set its configuration fully but this package also come with a set of preset configuration to answer most usecases
+// e.g. to create a client which performs exponential backoff and listens to `Retry-After` headers, the following can be done:
```
Comment (SamCHogg) [inline utils/http/interfaces.go:10]: Will these show as examples in godoc? For reference I found this: https://pkg.go.dev/github.com/fluhus/godoc-tricks (https://github.com/fluhus/godoc-tricks)

Comment (acabarbaye) [reply utils/http/interfaces.go:10]: I followed this https://github.com/fluhus/godoc-tricks/blob/1266307515d621e91d7972ddf41b003c6838219f/doc.go#L117

Comment (acabarbaye) [reply utils/http/interfaces.go:10]: I added the extra indent as suggested

**Review by SamCHogg — APPROVED** [review]

**Pull request merged**